src/Teacher.js

```diff
@@ -42,25 +43,65 @@ function GenerateTeacherPageHead()
     return "<head></head>";
 }
 
+function GenerateOverviewRow(assessment)
+{
+    return "<tr><th>" + assessment.GetName() + "</th><td>" + assessment.GetWeight() 
+        + "</td><td>" + assessment.GetAverage() + "</td></tr>";
+}
+
+function GenerateOverview(assessments)
+{
+    let htmlStr = "<table><caption><b>Overview</caption><tr><th></th><th>Weight</th><th>Average</th></tr>";
+    assessments.forEach(function (assessment)
+    {
+        htmlStr += GenerateOverviewRow(assessment);
+    });
+    return htmlStr;
+}
+
 function GenerateTeacherBody(user)
 {
     let body = "<body>";
     let assessments = user.GetAssessmentsTeacher();
+    
+    body += GenerateHeader();
+    //overview
+    //body += GenerateOverview(assessments);
+
+    // specifics
     assessments.forEach(function (assessment)
     {
         body += AssessmentToHTMLStrTeacher(assessment);
     });
+
+    body += GenerateFooter();
     body += "</body>";
     return body;
 }
 
+function GenerateStyle()
+{
+    return "<style> table, th, td {border:1px solid black;} </style>"
```

Comment:
style code should be in a css file that is referenced in the head of the webpage. For now that's ok but we will need to change this later.